### ISSUE TITLE
Requests with X-Access-Token and without user authorization returns 500 error

### ISSUE THREAD

**Issue by jasson99** [owncloud/ocis#1040]
## Steps to reproduce

Create a Post request such as : `curl --header "Tus-Resumable: 1.0.0" --header "Upload-Checksum: sha256 NjI2ZjZjOTk5OTNkYWE3ZmQxMWM0NmNiODY1ZmU5NTYyNWQwOTNmMzdiZWIyMjQ5OTUzMDQxOGRkNjMzOWNlYQ==" -H 'X-Access-Token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjA3NDk1NTExLCJpYXQiOjE2MDc0OTU0NTEsImlzcyI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJ1c2VyIjp7ImlkIjp7ImlkcCI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJvcGFxdWVfaWQiOiJBbGljZSJ9LCJ1c2VybmFtZSI6IkFsaWNlIiwibWFpbCI6ImFsaWNlQGV4YW1wbGUub3JnIiwibWFpbF92ZXJpZmllZCI6dHJ1ZSwiZGlzcGxheV9uYW1lIjoiQWxpY2UgSGFuc2VuIiwiZ3JvdXBzIjpbInVzZXJzIl0sIm9wYXF1ZSI6eyJtYXAiOnsiZ2lkIjp7ImRlY29kZXIiOiJwbGFpbiIsInZhbHVlIjoiTXpBd01EQT0ifSwicm9sZXMiOnsiZGVjb2RlciI6Impzb24iLCJ2YWx1ZSI6Ild5SmtOMkpsWldWaE9DMDRabVkwTFRRd05tSXRPR1ppTmkxaFlqSmtaRGd4WlRaaU1URWlYUT09In0sInVpZCI6eyJkZWNvZGVyIjoicGxhaW4iLCJ2YWx1ZSI6Ik1qQXdNRFU9In19fX19.KKatOqisI1z614OeRNybhrAfMlVb2H3wbewl0Uwwzgc' --header "Upload-Key: 1272017606" --header "Upload-Length: 16" --header "Upload-Metadata: filename L25vbkV4aXN0ZW50Rm9sZGVyL3RleHRmaWxlLnR4dA==" --user-agent "GuzzleHttp/7" --header "X-Forwarded-For: 127.0.0.1" --data "uploaded content" https://localhost:9200/remote.php/webdav/ -v -k`

The response is : `HTTP/1.1 500 Internal Server Error`

Now, create a Post request as above but adding user authorization and then a patch request as follows:

- Post: `curl --header "Tus-Resumable: 1.0.0" --user 'admin:admin' --header "Upload-Checksum: sha256 NjI2ZjZjOTk5OTNkYWE3ZmQxMWM0NmNiODY1ZmU5NTYyNWQwOTNmMzdiZWIyMjQ5OTUzMDQxOGRkNjMzOWNlYQ==" -H 'X-Access-Token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjA3NDk1NTExLCJpYXQiOjE2MDc0OTU0NTEsImlzcyI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJ1c2VyIjp7ImlkIjp7ImlkcCI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJvcGFxdWVfaWQiOiJBbGljZSJ9LCJ1c2VybmFtZSI6IkFsaWNlIiwibWFpbCI6ImFsaWNlQGV4YW1wbGUub3JnIiwibWFpbF92ZXJpZmllZCI6dHJ1ZSwiZGlzcGxheV9uYW1lIjoiQWxpY2UgSGFuc2VuIiwiZ3JvdXBzIjpbInVzZXJzIl0sIm9wYXF1ZSI6eyJtYXAiOnsiZ2lkIjp7ImRlY29kZXIiOiJwbGFpbiIsInZhbHVlIjoiTXpBd01EQT0ifSwicm9sZXMiOnsiZGVjb2RlciI6Impzb24iLCJ2YWx1ZSI6Ild5SmtOMkpsWldWaE9DMDRabVkwTFRRd05tSXRPR1ppTmkxaFlqSmtaRGd4WlRaaU1URWlYUT09In0sInVpZCI6eyJkZWNvZGVyIjoicGxhaW4iLCJ2YWx1ZSI6Ik1qQXdNRFU9In19fX19.KKatOqisI1z614OeRNybhrAfMlVb2H3wbewl0Uwwzgc' --header "Upload-Key: 1272017606" --header "Upload-Length: 16" --header "Upload-Metadata: filename L25vbkV4aXN0ZW50Rm9sZGVyL3RleHRmaWxlLnR4dA==" --user-agent "GuzzleHttp/7" --header "X-Forwarded-For: 127.0.0.1" --data "uploaded content" https://localhost:9200/remote.php/webdav/ -v -k`

- Response: 
``` 
HTTP/1.1 201 Created
Location: https://localhost:9200/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjU5MzU0MjQyLCJpYXQiOjE2MDc1MTQyNDIsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1NS9kYXRhL3R1cy8yYWY1ZDE0Zi05YmUyLTQ4ODktYjJjYS1lYTNlOWYyNzI2YzMifQ.zzHHnHzvgETieLfI0rBOHwxV4P5ms5PrW_LOXATyyVc

```

- Patch: `curl -XPATCH "https://localhost:9200/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjU5MzU0MjQyLCJpYXQiOjE2MDc1MTQyNDIsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1NS9kYXRhL3R1cy8yYWY1ZDE0Zi05YmUyLTQ4ODktYjJjYS1lYTNlOWYyNzI2YzMifQ.zzHHnHzvgETieLfI0rBOHwxV4P5ms5PrW_LOXATyyVc" -H 'Content-Length: 16' -H 'X-Access-Token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjA3NDk1NTExLCJpYXQiOjE2MDc0OTU0NTEsImlzcyI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJ1c2VyIjp7ImlkIjp7ImlkcCI6Imh0dHBzOi8vbG9jYWxob3N0OjkyMDAiLCJvcGFxdWVfaWQiOiJBbGljZSJ9LCJ1c2VybmFtZSI6IkFsaWNlIiwibWFpbCI6ImFsaWNlQGV4YW1wbGUub3JnIiwibWFpbF92ZXJpZmllZCI6dHJ1ZSwiZGlzcGxheV9uYW1lIjoiQWxpY2UgSGFuc2VuIiwiZ3JvdXBzIjpbInVzZXJzIl0sIm9wYXF1ZSI6eyJtYXAiOnsiZ2lkIjp7ImRlY29kZXIiOiJwbGFpbiIsInZhbHVlIjoiTXpBd01EQT0ifSwicm9sZXMiOnsiZGVjb2RlciI6Impzb24iLCJ2YWx1ZSI6Ild5SmtOMkpsWldWaE9DMDRabVkwTFRRd05tSXRPR1ppTmkxaFlqSmtaRGd4WlRaaU1URWlYUT09In0sInVpZCI6eyJkZWNvZGVyIjoicGxhaW4iLCJ2YWx1ZSI6Ik1qQXdNRFU9In19fX19.KKatOqisI1z614OeRNybhrAfMlVb2H3wbewl0Uwwzgc' -H 'Tus-Resumable: 1.0.0' -H 'Content-Type: application/offset+octet-stream' -H 'Upload-Offset: 0' --data "uploaded content" -v -k
`

- Response: `HTTP/1.1 500 Internal Server Error`

**Comment by individual-it** [comment]
does this only happen for TUS requests or also other requests?

**Comment by individual-it** [comment]
or does this only happen when we run ocis with `PROXY_ENABLE_BASIC_AUTH=true` ?

**Comment by refs** [comment]
ping @jasson99 , could you address @individual-it comments? :)

**Comment by jasson99** [comment]
This request returns 401 now. It seems to be fixed after this commit: 92a1bc8fb68a2ab20ecf13fdef6ca3ddbc4ec561
```
92a1bc8fb68a2ab20ecf13fdef6ca3ddbc4ec561 is the first bad commit
commit 92a1bc8fb68a2ab20ecf13fdef6ca3ddbc4ec561
Author: Ilja Neumann <ineumann@owncloud.com>
Date:   Fri Dec 4 01:56:54 2020 +0100

    Make it possible to use CS3 as accounts backend instead of account-service
    
    Configureable via:
    PROXY_ACCOUNT_BACKEND_TYPE=cs3
    PROXY_ACCOUNT_BACKEND_TYPE=accounts (default)
    
    By using a backend which implements the CS3 user-api (currently provided by reva/storage) it is possible to bypass
    the ocis-accounts service and for example use ldap directly.
    
    Hides user and auth related communication behind a facade (user/backend) to minimize logic-duplication across middlewares.
    Allows to switich the account backend from accounts to cs3.
    
    Co-authored-by: Jörn Friedrich Dreyer <jfd@butonic.de>

:040000 040000 3383cfd12e36ef8adf214320c2f4be743eeef4cf 2c0041ec92ab89b779a10ad1dccba701783e8ace M	changelog
:040000 040000 3d8a72d1d60a90209d9dd5a832c5b14b236b1c22 0c7bcf933adbe76dc8a7f3ea2b40005727e46c7a M	proxy
```
